number_theory/modular.h

```diff
@@ -0,0 +1,82 @@
+#ifndef NUMBER_THEORY_MODULAR_H_
+#define NUMBER_THEORY_MODULAR_H_
+
+#include <stddef.h>
+
+#include <bit>
+#include <limits>
+
+#include "number_theory/utility.h"
+
+// Modular arithmetic
+
+namespace tql {
+namespace number_theory {
+
+template <class T, T mod>
```

Comment:
What about making `int64_t` or `uint64_t` default for `T`? I remember seeing tourist's templates. His type was fixed to `long long` so he was using it like `using Mint = Modular<mod>;`. I suggest either removing the type parameter or defining a default so that the usage can be simple like tourist's version.

Reply:
We can't set the default type for the first argument like `template<class T=int, T mod>`, because C++ language only allows default values for trailing arguments.